catalyst_voices/packages/internal/catalyst_voices_blocs/lib/src/proposal/proposal_cubit.dart

```diff
@@ -6,68 +6,169 @@ import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
-    String? message,
-  }) {
-    final commentTemplate = DocumentSchema.optional(
-      properties: [
-        DocumentGenericObjectSchema.optional(
-          nodeId: DocumentNodeId.fromString('comment'),
-          description: const MarkdownData('The comments on the proposal'),
-          properties: [
-            DocumentMultiLineTextEntrySchema.optional(
-              nodeId: DocumentNodeId.fromString('comment.content'),
-              description: const MarkdownData('The comment text content'),
-              strLengthRange: const Range(min: 1, max: 5000),
-            ),
-          ],
-        ),
-      ],
-    );
+  Future<void> load({required DocumentRef ref}) async {
+    try {
+      _logger.info('Loading $ref');
+
+      emit(state.copyWith(isLoading: true));
 
-    final builder = DocumentBuilder.fromSchema(schema: commentTemplate);
+      final proposal = await _proposalService.getProposal(ref: ref);
+      final comments = _buildComments();
 
-    if (message != null) {
-      final change = DocumentValueChange(
-        nodeId: DocumentNodeId.fromString('comment.content'),
-        value: message,
+      if (isClosed) {
+        return;
+      }
+
+      final commentsSort = state.data.commentsSort;
+
+      final proposalViewData = _buildProposalViewData(
+        hasActiveAccount: _userService.user.activeAccount != null,
+        proposal: proposal,
+        comments: comments,
+        commentSchema: _buildCommentTemplate().document.schema,
+        commentsSort: commentsSort,
+        isFavorite: false,
       );
-      builder.addChange(change);
-    }
 
-    final document = builder.build();
+      _ref = ref;
+      _proposal = proposal;
+      _comments = comments;
+
+      emit(ProposalState(data: proposalViewData));
+
+      if (proposalViewData.isCurrentVersionLatest == false) {
+        emitSignal(const ViewingOlderVersionSignal());
+      }
+    } on LocalizedException catch (error, stack) {
+      _logger.severe('Loading $ref failed', error, stack);
+
+      _ref = null;
+      _proposal = null;
+      _comments = null;
+
+      emit(ProposalState(error: error));
+    } catch (error, stack) {
+      _logger.severe('Loading $ref failed', error, stack);
+
+      _ref = null;
+      _proposal = null;
+      _comments = null;
+
+      emit(const ProposalState(error: LocalizedUnknownException()));
+    } finally {
+      emit(state.copyWith(isLoading: false));
+    }
+  }
 
-    return CommentDocument(
+  Future<void> submitComment({
+    required Document document,
+    SignedDocumentRef? parent,
+  }) async {
+    final comment = CommentDocument(
       metadata: CommentMetadata(
-        selfRef: selfRef ?? SignedDocumentRef.generateFirstRef(),
+        selfRef: SignedDocumentRef.generateFirstRef(),
         parent: parent,
       ),
       document: document,
     );
+
+    final data = state.data;
+
+    final updatedData = data.addComment(comment);
+
+    emit(state.copyWith(data: updatedData));
```

Comment:
How about keeping related things next to each other?

```suggestion
    final data = state.data;
    final updatedData = data.addComment(comment);
    emit(state.copyWith(data: updatedData));
```